lib/Report/SummaryReport.php

```diff
@@ -0,0 +1,232 @@
+        if (!file_exists($zipFile))
+            throw new \InvalidArgumentException(__('File does not exist'));
+
+        // Open the Zip file
+        $zip = new \ZipArchive();
+        if (!$zip->open($zipFile))
+            throw new \InvalidArgumentException(__('Unable to open ZIP'));
+
+        // Get the reportscheduledetails
+        $results = json_decode($zip->getFromName('reportschedule.json'), true);
+
+        // Return data to build chart
+        return [
+            'template' => 'summary-report-preview',
+            'saveAs' => $savedReport->saveAs,
+            'labels' => json_encode($results['labels']),
+            'countData' => json_encode($results['countData']),
+            'durationData' => json_encode($results['durationData']),
+            'backgroundColor' => json_encode($results['backgroundColor']),
+            'borderColor' => json_encode($results['borderColor']),
+        ];
+    }
+
+    /** @inheritdoc */
+    public function getResults($filterCriteria)
+    {
+
+        $fromDt = $this->getSanitizer()->getDate('fromDt', $this->getSanitizer()->getDate('statsFromDt', $this->getDate()->parse()->addDay(-1)));
+        $toDt = $this->getSanitizer()->getDate('toDt', $this->getSanitizer()->getDate('statsToDt', $this->getDate()->parse()));
+
+        $type = strtolower($this->getSanitizer()->getString('type', $filterCriteria));
+        $layoutId = $this->getSanitizer()->getInt('layoutId', $filterCriteria);
+        $mediaId = $this->getSanitizer()->getInt('mediaId', $filterCriteria);
+        $reportFilter = $this->getSanitizer()->getString('reportFilter', $filterCriteria);
+        $groupByFilter = $this->getSanitizer()->getString('groupByFilter', $filterCriteria);
+
+        // What if the fromdt and todt are exactly the same?
+        // in this case assume an entire day from midnight on the fromdt to midnight on the todt (i.e. add a day to the todt)
+        if ($fromDt == $toDt) {
+            $toDt->addDay(1);
+        }
+
+        $diffInDays = $toDt->diffInDays($fromDt);
+
+        // Format param dates
+        $fromDt = $this->getDate()->getLocalDate($fromDt);
+        $toDt = $this->getDate()->getLocalDate($toDt);
+
+        // Get an array of display id this user has access to.
+        $displayIds = [];
+
+        foreach ($this->displayFactory->query() as $display) {
+            $displayIds[] = $display->displayId;
+        }
+
+        if (count($displayIds) <= 0)
+            throw new InvalidArgumentException(__('No displays with View permissions'), 'displays');
+
+        $labels = [];
+        $countData = [];
+        $durationData = [];
+        $backgroundColor = [];
+        $borderColor = [];
+
+        // Call the time series interface get daily summary report
+        $result =  $this->getTimeSeriesStore()->getDailySummaryReport($displayIds, $diffInDays, $type, $layoutId, $mediaId, $reportFilter, $groupByFilter, $fromDt, $toDt);
```

Comment:
So this bit is interesting - we should discuss!
Having our clever abstract report classes doesn't help if we eventually have to put all the report methods in the time series stores.